BlueprintUI/Sources/Layout/ConstrainedPercent.swift

```diff
@@ -0,0 +1,85 @@
+    /// `nil` if the height is unconstrained.
+    public var height: CGFloat?
+
+    /// Initializes with the given properties.
+    ///
+    /// - parameters:
+    ///   - width: The percentage of the parent's width the content element should occupy. Values between 0-1
+    ///   correspond to 0-100%. `nil` if the width is unconstrained. By default, `nil`.
+    ///   - height: The percentage of the parent's height the content element should occupy. Values between 0-1
+    ///   correspond to 0-100%. `nil` if the height is unconstrained. By default, `nil`.
+    ///   - wrapping: The content element.
+    public init(width: CGFloat? = nil, height: CGFloat? = nil, wrapping wrappedElement: Element) {
+        self.width = width
+        self.height = height
+        self.wrappedElement = wrappedElement
+    }
+
+    public var content: ElementContent {
+        ElementContent(
+            child: wrappedElement,
+            layout: Layout(width: width, height: height)
+        )
+    }
+
+    public func backingViewDescription(bounds: CGRect, subtreeExtent: CGRect?) -> ViewDescription? {
+        return nil
+    }
+
+    private struct Layout: SingleChildLayout {
+        var width: CGFloat?
+        var height: CGFloat?
+
+        func measure(in sizeConstraint: SizeConstraint, child: Measurable) -> CGSize {
+            /// Get the target layout space, constraining it by the specified percentages.
+            let targetWidth = width.map { $0 * sizeConstraint.width.maximum }
+            let targetHeight = height.map { $0 * sizeConstraint.height.maximum }
+
+            let constraint = SizeConstraint(
+                width: targetWidth.map { .atMost($0) } ?? .unconstrained,
+                height: targetHeight.map { .atMost($0) } ?? .unconstrained
+            )
+
+            /// Measure the child in the constrained layout space.
+            let measurement = child.measure(in: constraint)
+
+            /// Override the child's measurement with a target dimension if one exists.
+            return CGSize(
+                width: targetWidth ?? measurement.width,
+                height: targetHeight ?? measurement.height
+            )
+        }
+
+        func layout(size: CGSize, child: Measurable) -> LayoutAttributes {
+            return LayoutAttributes(size: size)
+        }
+    }
+}
+
+
+public extension Element {
+    /// Constrains the element to a percentage of its parent's size.
+    ///
+    /// - parameters:
+    ///   - widthPercent: The percentage of the parent's width the content element should occupy. Values between 0-1
+    ///   correspond to 0-100%. `nil` if the width is unconstrained. By default, `nil`.
+    ///   - heightPercent: The percentage of the parent's height the content element should occupy. Values between 0-1
+    ///   correspond to 0-100%. `nil` if the height is unconstrained. By default, `nil`.
+    func constrainedTo(
+        widthPercent: CGFloat? = nil,
```

Comment:
+1 to fraction from @kylebshr

Reply:
Updated `ConstrainedPercent` to `ConstrainedSizeFraction`. This site became `widthFraction` in the shift.